### DISCUSSION TITLE
Problems with xpkg.upbound.io mirroring in Jfrog artifactory

### DISCUSSION THREAD

**Discussion by mleahu** [crossplane/crossplane#5413] [General]
Hi

We use an internal Jfrog artifactory to mirror the external artifactories as well and we started to mirror the newly xpkg.upbound.io for crossplane 1.14.5 and 1.15.0 . Until now we mirrored the index.docker.io and everything worked well.

Now, with xpkg.upbound.io mirror, we started to encounter issues like:
 - initial pull of image with docker pull cli stops in the middle of layers download with the error `error pulling image configuration: download failed after attempts=1: unknown blob`. Then, about an hour later, another docker pull would work just fine.
 - even if the download succeeds, there is a warning shown by docker cli. This reads:
`
WARNING: ⚠️ Failed to pull manifest by the resolved digest. This registry does not
appear to conform to the distribution registry specification; falling back to
pull by tag. This fallback is DEPRECATED, and will be removed in a future
release. ⚠️
`
The [documentation on Jfrog site](https://jfrog.com/help/r/artifactory-how-to-deal-with-failed-to-pull-manifest-by-the-resolved-digest-this-registry-does-not-appear-to-conform-to-the-distribution-registry-specification-falling-back-to-pull-by-tag-when-proxying-docker-registries-via-artifactory-using-docker-e) asks the artifactory admins to migrate images from v1 to v2.
 
Has anyone else encountered the same issues?

Thank you!

**Comment by jbw976** [comment]
Thanks for sharing your findings here @mleahu - does this happen with specific packages/repos, or does it affect all equally? Do you have a specific example of a package/repo that you are seeing this behavior with?

**Reply by mleahu** [reply]
The crossplane packages are showing this behavior.

For example:
```
docker pull xpkg-upbound-docker-remote.artifactory.xxxxxxxxxxx.com/crossplane/crossplane:v1.15.0
v1.15.0: Pulling from crossplane/crossplane
07a64a71e011: Already exists 
fe5ca62666f0: Already exists 
280126c0e181: Downloading [>                                                  ]  7.207kB/717.1kB
fcb6f6d2c998: Download complete 
e8c73c638ae9: Download complete 
1e3d9b7d1452: Waiting 
4aa0ea1413d3: Waiting 
7c881f9ab25e: Waiting 
5627a970d25e: Waiting 
8ac2e1270e61: Waiting 
b4c1d679c19f: Waiting 
ea2fcf9ad855: Waiting 
error pulling image configuration: download failed after attempts=1: unknown blob
```
then later, after about an hour, it succeeds.

Also, it happened for crossplane 1.14.5 to have the warning from the initial message.

Thank you!

**Comment by djeremiah** [comment]
Somewhat related, though not directly:
Mirroring using docker pull->docker push is not a particularly good approach, particularly for provider/configuration/
function packages, because hashes may be recomputed when the image is unpacked on the local machine, resulting in fallback behavior from the XP package manager and much slower provider deployment. Better to copy directly using a tool like skopeo or crane.
`skopeo copy --all --preserve-digests docker://xpkg.upbound.io/upbound/provider-aws-s3:v0.41.0 docker://artifactory-repo/upbound/provider-aws-s3:v0.41.0`
You could use the same approach for crossplane itself. It may be that use of one of these tools sidesteps the issue you're seeing.

**Comment by RahulPandey061192** [comment]
hey @djeremiah 
Thanks for your suggestions.
I mirrored the aws images using below command:
crane copy -a  xpkg.upbound.io/upbound/provider-aws-sqs *private-repo-url*
I copied all the versions of the providers for me .

But when i am using the provider with crossplane to deploy by claims it is giving me below error:
**Timeout: failed waiting for *v1beta1.ProviderConfig Informer to sync**

Any suggestions please??
